tests/components/control4/test_climate.py

```diff
@@ -0,0 +1,502 @@
+@pytest.mark.usefixtures("mock_c4_account", "mock_c4_director")
+async def test_climate_cool_state(
+    hass: HomeAssistant,
+    mock_config_entry,
+    platforms,
+) -> None:
+    """Test climate entity in cool state."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    mock_config_entry.add_to_hass(hass)
+
+    climate_vars = {
+        123: {
+            "HVAC_STATE": "cooling",
+            "HVAC_MODE": "Cool",
+            "TEMPERATURE_F": 74.0,
+            "HUMIDITY": 55,
+            "COOL_SETPOINT_F": 72.0,
+            "HEAT_SETPOINT_F": 68.0,
+        }
+    }
+
+    async def mock_update_vars(*args, **kwargs):
+        return climate_vars
+
+    with (
+        patch(
+            "homeassistant.components.control4.climate.update_variables_for_config_entry",
+            new=mock_update_vars,
+        ),
+        patch("homeassistant.components.control4.PLATFORMS", platforms),
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == HVACMode.COOL
+    assert state.attributes["hvac_action"] == HVACAction.COOLING
+    assert state.attributes["temperature"] == 72.0
+
+
+@pytest.mark.usefixtures("mock_c4_account", "mock_c4_director")
+async def test_climate_auto_state(
+    hass: HomeAssistant,
+    mock_config_entry,
+    platforms,
+) -> None:
+    """Test climate entity in auto state."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    mock_config_entry.add_to_hass(hass)
+
+    climate_vars = {
+        123: {
+            "HVAC_STATE": "heating",
+            "HVAC_MODE": "Auto",
+            "TEMPERATURE_F": 65.0,
+            "HUMIDITY": 40,
+            "COOL_SETPOINT_F": 75.0,
+            "HEAT_SETPOINT_F": 68.0,
+        }
+    }
+
+    async def mock_update_vars(*args, **kwargs):
+        return climate_vars
+
+    with (
+        patch(
+            "homeassistant.components.control4.climate.update_variables_for_config_entry",
+            new=mock_update_vars,
+        ),
+        patch("homeassistant.components.control4.PLATFORMS", platforms),
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == HVACMode.HEAT_COOL
+    assert state.attributes["hvac_action"] == HVACAction.HEATING
+    assert state.attributes["target_temp_high"] == 75.0
+    assert state.attributes["target_temp_low"] == 68.0
+
+
+@pytest.mark.usefixtures(
+    "mock_c4_account", "mock_c4_director", "mock_climate_update_variables"
+)
+async def test_set_hvac_mode(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test setting HVAC mode."""
+    hass.config.units = US_CUSTOMARY_SYSTEM
+    await setup_integration(hass, mock_config_entry)
+
+    with patch(
+        "pyControl4.climate.C4Climate.setHvacMode", new_callable=AsyncMock
+    ) as mock_set:
```

Comment:
1. Let's patch this where we use it
2. Why don't we have a more global patch for C4Climate?